src/backend/optimizer/plan/planner.c

```diff
@@ -576,6 +577,27 @@ subquery_planner(PlannerGlobal *glob, Query *parse,
 	parse->jointree = (FromExpr *)
 		pull_up_subqueries(root, (Node *) parse->jointree, false, false);
 
+	if ((parent_root && parent_root->is_correlated_subplan) ||
+		((Gp_role == GP_ROLE_DISPATCH) &&
+		root->config->is_under_subplan &&
+		IsSubqueryCorrelated(parse)))
+	{
+		root->is_correlated_subplan = true;
+		/*
+		 * Generate the plan for the subquery with certain options disabled.
+		 */
+		config->gp_enable_direct_dispatch = false;
+		config->gp_enable_multiphase_agg = false;
+
+		/*
+		 * The MIN/MAX optimization works by inserting a subplan with LIMIT 1.
+		 * That effectively turns a correlated subquery into a multi-level
+		 * correlated subquery, which we don't currently support. (See check
+		 * above.)
+		 */
+		// config->gp_enable_minmax_optimization = false;
```

Comment:
Why do we need this commented code? I have recollected a very interesting [koan](http://thecodelesscode.com/case/41) about commented code...

Reply:
It was my mistake. I deleted that code.